src/image/image_type.cpp

```diff
@@ -7,6 +7,19 @@
 
 namespace turi{
 
+image_type::image_type(boost::gil::rgb8_image_t gil_image) {
+  auto it = view(gil_image).begin();
+  auto data = (uint8_t*) &boost::gil::at_c<0>(*it);
+  m_height = gil_image.height();
```

Comment:
These can be moved to the initializers where possible -- do 
```
image_type::image_type(boost::gil::rgb8_image_t gil_image) 
: m_height(gil_image.height())
, m_width(gil_image.width())
...
{
/// Just the data copy here
```

Reply:
I rearchitected the traditional image_type constructor to use initializers, and made the constructor from the boost::gil image just call that. How does that sound?